crates/plasma/Cargo.toml

```diff
@@ -15,7 +15,7 @@ tracing.workspace = true
 
 # Local
 kona-primitives = { path = "../primitives" }
-kona-providers = { path = "../providers" }
+kona-derive = { path = "../derive" }
```

Comment:
I'd like to avoid importing `kona-derive` in `kona-plasma`. Feels leaky.

Reply:
But `kona-plasma` is an extension of `kona-derive`?

Reply:
Discussed offline - this is the dependency graph we will go with for now. Since the dependency is not a component but a trait, I think it's reasonable not to refactor into a third node in the dependency graph, concretely a new crate, and add edges from both `kona-derive` and `kona-plasma` to that third node (eg `kona-providers`).